src/providers/latest-telemetry-provider.js

```diff
@@ -119,9 +121,14 @@ export default class LatestTelemetryProvider {
         this.#bulkPromise = undefined;
     }
 
-    #waitOneEventCycle() {
+    #waitForDebounce() {
```

Comment:
I had completely forgotten that batching was already implemented! The previous method did not use true debouncing, consequently the way that this function is called will never truly debounce the call either. I think it will _always_ wait 100ms no matter what, right? Because we are basically using the presence of `this.#bulkPromise` as a semaphore.

It's fine how it is for now. We _might_ be able to improve this in future by removing the semaphore and having it always call the debounce function each time, which will keep clearing the timeout until it's settled down. We could then probably reduce or even remove the 100ms wait.

Reply:
Sounds good!